### PR TITLE
Create authenticator directly

### PR DESCRIPTION
**Pull request by aronerben**

### CODE REVIEW

File: lib/letters.ml [hunk 1]
```diff
@@ -176,9 +176,9 @@ let ca_path_peer_verifier path =
   Lwt.return (X509.Authenticator.chain_of_trust ~time:now certs)
 ;;
 
-(* Store the detected CA cert path so it doesn't have to be detected every time a mail is
-   sent *)
-let detected_cert = ref None
+(* Store the auto-detected CA cert authenticator so it doesn't have to be detected every
+   time a mail is sent *)
+let detected_auth = ref None
```
Comment (mikonieminen) [inline lib/letters.ml]: This is pretty much same as using global variable and should be fixed in a future release where this is encapsulated properly.

Comment (aronerben) [reply lib/letters.ml]: I wrapped it into the lexical context of a closure. The new diff git shows is horrendous. For the closure, all I did was adding the following lines to the top of `send`:
```ocaml
  (* Store the auto-detected CA cert authenticator so it doesn't have to be detected every
     time a mail is sent *)
  let detected_auth = ref None in
  fun ~config:c ~sender ~recipients ~message ->
  ```

**Comment by mikonieminen** [comment]
I rebased and force pushed. I also updated opam lockfile.

**Review by mikonieminen — APPROVED** [review]

**Comment by aronerben** [comment]
> I rebased and force pushed. I also updated opam lockfile.

Thanks, I tested it and it works. However, as mentioned, a warning is shown on my system:
```
2021-09-08T10:29:36-00:00 [WARNING] [ca-certs]: Failed to decode a trust anchor No certificate.
```
I assume it's because one of the PEM files on my system is not a full certificate.

**Comment by mikonieminen** [comment]
> > I rebased and force pushed. I also updated opam lockfile.
> 
> Thanks, I tested it and it works. However, as mentioned, a warning is shown on my system:
> 
> ```
> 2021-09-08T10:29:36-00:00 [WARNING] [ca-certs]: Failed to decode a trust anchor No certificate.
> ```
> 
> I assume it's because one of the PEM files on my system is not a full certificate.

I don't see this warning and sounds like it might be the case you suggested. So I think this can be ignored.

**Review by mikonieminen — APPROVED** [review]

**Pull request merged**